bin/transform

```diff
@@ -95,13 +95,22 @@ def parse_geographic_columns(gisaid_data: pd.DataFrame) -> pd.DataFrame:
     Expands the string found in the column named `location` in the given
     *df*, creating four new columns. Returns the modified ``pd.DataFrame``.
     """
-    geographic_data = gisaid_data['location'].str.split('\s*/\s*', expand=True)
+    geographic_data = gisaid_data['location'].str.split(r'\s*/\s*', expand=True)
+    location_columns = ['region', 'country', 'division', 'location']
 
-    gisaid_data['region']      = geographic_data[0].str.strip()
-    gisaid_data['country']     = geographic_data[1].str.strip()
-    gisaid_data['division']    = geographic_data[2].str.strip()
-    gisaid_data['location']    = geographic_data[3].str.strip()
+    for column, index in zip(location_columns, range(len(location_columns))):
+        gisaid_data[column] = geographic_data[index].str.replace('_', ' ').str.strip()
 
+        not_null = pd.notnull(gisaid_data[column])
+        is_uppercase = gisaid_data[column].str.upper() == gisaid_data[column]
```

Comment:
I'm not entirely sure this is what we want? I assume we want to convert all locations to title case to standardize names. For example, in the live site, there's currently `New Taipei city` and `New Taipei City`.

Reply:
Would it be weird if everything was just in uppercase?

Reply:
I see the following non-mutually exclusive programmatic options:
* we could use a third-party library like `titlecase` that handles English articles
* we could "whitelist" non-English articles like "de", "la", "do", etc. 
* we could whitelist US state abbreviations like DC, WA, WI that appear in location names

Locations that slip through could continue to be manually annotated or added to the whitelist. Which seems like more work -- the status quo of annotating locations or fixing improperly title cased locations (either in annotations.tsv or in this script)?

Reply:
I think it's worthwhile to standardize as much as possible within the script and manually annotate things that slip through. If a certain pattern is seen often enough, then we add it to the hardcoded whitelist?

Reply:
I think it will be less work to have as much as possible be standardized in the script, and fixing things that slip through manually. I think that would still be an improvement over the status quo. To be honest, I am most excited about the underscore replacement and I did not give adequate thought to how complicated the casing is.

Reply:
@lmoncla @joverlee521 I opened #59 to be able to immediately merge the underscore replacement while we continue working/thinking through title casing on this PR.

Reply:
@joverlee521 Ok, I have reverted my old title casing commits in favor of a new one. 
This one uses `.title()` because the benefit of using the third party `titlecase` library is that it uses the context of the whole string. For example, 
```
>>> from titlecase import titlecase
>>> titlecase('of')
'Of'
>>> titlecase('united stats of america')
'United Stats of America'
```

Because `titlecase` does not work with non-English articles (de, la, etc.), it was necessary to re-process any `titlecase`-produced strings either by tokenization and iteration or string substitution to set the articles back to their correct capitalization. Therefore, I opted in favor of capitalizing each token (i.e. word) separately to process each string only once. This reduced the advantage of `titlecase` which is why I instead used `.title()`. 

I also found another edge case for Francophone locations where hyphen-delimited locations (e.g. Braine-le-Compte) don't register as separate tokens. However all parts do get capitalized by `.title()`, so I wrote a separate regex that looks for this pattern and prevents manipulation on it.

Reply:
I'm open to criticism here if we decide we do want to process entire resolution name strings with `titlecase` then retroactively substitute articles back to their correct capitalization.